CI failures with pytest 8

For a few days, CI has been failing consistently. Different platforms, different molecular I/O library, different network libraries.

The most notable difference between the last green CI and the failing CI is `pytest`, gone from `7.4.4` to `8.0.0`.

Looking at the failure, it appears that the error message raised and the one captured by `pytest` do not match:
```
E       Failed: DID NOT WARN. No warnings of type (<class 'UserWarning'>,) matching the regex were emitted.
E        Regex: No atomic number information stored on nodes.
E        Emitted warnings: [ UserWarning('No atomic property information stored on nodes. Node matching is not performed...')].
```

It appears that some of the error messages were not updated in #64. However, this failed silently because of https://github.com/pytest-dev/pytest/issues/9288.

Since `pyrest=8.0.0`, [`warns()` [...] re-emits unmatched warnings when the context closes – previously it would consume all warnings, hiding those that were not matched by the function](https://docs.pytest.org/en/stable/changelog.html#other-breaking-changes).

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -8,6 +8,10 @@
 Date:            XX/YY/ZZZZ
 Contributors:    @RMeli
 
+### Fixed
+
+* Failing tests with `pytest=8.0.0` [PR #101 | @RMeli]
+
 ### Improved
 
 * Messages for `NotImplementedError` exceptions [PR #90 | @RMeli]

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -4,6 +4,7 @@ Control skipping of tests according to command line option.
 Soource:
     https://docs.pytest.org/en/latest/example/simple.html
 """
+
 import numpy as np
 import pytest
 

--- a/tests/test_benchmarks.py
+++ b/tests/test_benchmarks.py
@@ -19,7 +19,6 @@ def systems():
     return systems
 
 
-@pytest.mark.benchmark
 @pytest.fixture(scope="module", params=systems())
 def molecules(request):
     system = request.param

--- a/tests/test_graph.py
+++ b/tests/test_graph.py
@@ -3,6 +3,7 @@ import pytest
 
 from spyrmsd import constants, graph, io, molecule
 from spyrmsd.exceptions import NonIsomorphicGraphs
+from spyrmsd.graphs import _common as gc
 from tests import molecules
 
 
@@ -104,9 +105,7 @@ def test_graph_from_adjacency_matrix_atomicnums(rawmol, mol) -> None:
     ],
 )
 def test_match_graphs_isomorphic(G1, G2) -> None:
-    with pytest.warns(
-        UserWarning, match="No atomic property information stored on nodes."
-    ):
+    with pytest.warns(UserWarning, match=gc.warn_no_atomic_properties):
         isomorphisms = graph.match_graphs(G1, G2)
 
     assert len(isomorphisms) != 0
@@ -121,8 +120,8 @@ def test_match_graphs_isomorphic(G1, G2) -> None:
 )
 def test_match_graphs_not_isomorphic(G1, G2) -> None:
     with pytest.raises(
-        NonIsomorphicGraphs, match="Graphs are not isomorphic."
-    ), pytest.warns(UserWarning, match="No atomic number information stored on nodes."):
+        NonIsomorphicGraphs, match=gc.error_non_isomorphic_graphs
+    ), pytest.warns(UserWarning, match=gc.warn_no_atomic_properties):
         graph.match_graphs(G1, G2)
 
 

